libcoap/src/session/mod.rs

```diff
@@ -296,6 +296,13 @@ pub trait CoapSessionCommon<'a>: CoapSessionCommonInternal<'a> {
         unsafe { coap_session_set_mtu(self.inner_mut().raw_session, mtu) }
     }
 
+    /// Sets maximum number of hops for multicast request
+    fn set_mcast_hops_limit(&self, hops: usize) {
+        // In some cases the default hoplimit of 1 is not enough to reach the destination. Choose limit accordingly
+        // SAFETY: Provided session pointer being valid is an invariant of CoapSessionInner
+        unsafe { coap_mcast_set_hops(self.inner_mut().raw_session, hops) };
```

Comment:
`coap_mcast_set_hops()` may fail. Please check the return value and return a `Result<()>`.

You may have to add an error type for that (or you could reuse `MulticastGroupJoinError` and rename it to `MulticastError` instead).